nvdaHelper/vbufBackends/mshtml/mshtml.cpp

```diff
@@ -90,9 +90,9 @@ template<typename toInterface> inline HRESULT queryService(IUnknown* pUnknown, c
 	}
 	hRes=pServProv->QueryService(siid,__uuidof(toInterface),(void**)pIface);
 	pServProv->Release();
-	if(hRes!=S_OK||!pIface) {
+	if(hRes!=S_OK) {

```

Comment:
Why remove the check for null pIface? I have experienced many implementations of QueryService and QueryInterface that may return S_OK yet leave pIface as null. Incorrect certainly, but it happens. I would certainly put this check back.

Reply:
No, then it will dereference a nullptr. See line 95. Note that we are passing in a `void**` if `pIface` is a `nullptr` the mistake is before the call to `QueryService`, right? Perhaps this code should have been:
```suggestion
	if( hRes != S_OK || *pIface == nullptr) {

```